state/action.go

```diff
@@ -331,15 +332,25 @@ func newAction(st *State, adoc actionDoc) Action {
 	}
 }
 
+// MinVersionSupportNewActionID should be un-exposed after 2.7 released.
+// TODO(action): un-expose MinVersionSupportNewActionID and IsNewActionIDSupported and remove those helper functions using these two vars in tests from 2.7.0.
+var MinVersionSupportNewActionID = version.MustParse("2.7.0")
```

Comment:
Why are we using a MinVersion check here? This is the reason we added Facade versioning. It should use the fact that the Uniter API is at an older version to maintain consistency, not a modelAgentVersion. If we need to pass a boolean from the Uniter API to indicate that it needs to do an old method of generating action identifiers, then it can do so.

But 

a) We should never remove the compatibility check until we actually drop compatibility with the *old* version of the agent. (eg, if we remove this, then when someone in the future upgrades a controller from 2.5 => 2.8 all of the model agents that aren't upgraded yet will break.)

b) We already have Versioned API facades to do this sort of check. Since we expect to do it at that level, that is more robust than doing it as a model agent version check.


Is there a reason this was done at the State layer instead?

Reply:
Ah, maybe the issue is that the client requesting the Action is 2.7, and the Controller is 2.7, but the agents that are watching for actions are 2.6 (thus a different API that is creating this vs the one that is consuming it).
That makes a bit more sense. But I'd like us to update the comment to indicate that it isn't about when 2.7 is released or not, but that it is compatibility with older unit agents.

Reply:
What I meant was to change MinVersionSupportNewActionID to `minVersionSupportNewActionID` after 2.7 released. 
It's exposed to the outside of the state package for passing some tests(The `currentVersion` is smaller than this min version which failed some new action ID related tests.).
Sorry for the confusion.